core/src/main/java/com/ttbmp/cinehub/core/usecase/buyticket/BuyTicketUseCaseController.java

```diff
@@ -0,0 +1,236 @@
+            }
+            if (Boolean.TRUE.equals(request.getFoldingArmchairOption())) {
+                ticket = new TicketFoldingArmchair(ticket);
+                ticket.setPrice(ticket.operation());
+            }
+            if (Boolean.TRUE.equals(request.getSkipLineRadioOption())) {
+                ticket = new TicketSkipLine(ticket);
+                ticket.setPrice(ticket.operation());
+            }
+            /*-----------------------------------------*/
+
+            ticket.setPosition(position);
+            buyTicketPresenter.setSelectedTicket(new GetTicketBySeatsResponse(TicketDataMapper.mapToDto(ticket)));
+        } catch (Request.NullRequestException e) {
+            buyTicketPresenter.presentGetTicketBySeatsNullRequest();
+        } catch (Request.InvalidRequestException e) {
+            buyTicketPresenter.presentInvalidGetTicketBySeats(request);
+        }
+
+
+    }
+
+
+    @Override
+    public void getListCinema(GetListCinemaRequest request) {
+        try {
+            Request.validate(request);
+            Movie movie = MovieDataMapper.mapToEntity(request.getMovieDto());
+            String date = request.getData();
+            List<Projection> projectionList = projectionRepository.getProjectionByMovie(movie, date);//devo passare anche la data
+            List<Cinema> cinemaList = cinemaRepository.getCinemaByProjection(projectionList);
+
+            buyTicketPresenter.presentCinemaList(new GetListCinemaResponse(CinemaDataMapper.mapToDtoList(cinemaList)));
+        } catch (Request.NullRequestException e) {
+            buyTicketPresenter.presentGetListCinemaNullRequest();
+        } catch (Request.InvalidRequestException e) {
+            buyTicketPresenter.presentInvalidGetListCinema(request);
+        }
+
+
+    }
+
+    @Override
+    public void getTimeOfProjection(GetTimeOfProjecitonRequest request) {
+        try {
+            Request.validate(request);
+            Cinema cinema = CinemaDataMapper.mapToEntity(request.getCinemaDto());
+            Movie movie = MovieDataMapper.mapToEntity(request.getMovieDto());
+            List<String> timeList = new ArrayList<>();
+            if (cinema != null) {
+                timeList = projectionRepository.getTimeOfProjectionByMovieCinema(movie, cinema);
+            }
+            buyTicketPresenter.presentTimeList(timeList);//Non serve una reponse per una lista di stringhe
+        } catch (Request.NullRequestException e) {
+            buyTicketPresenter.presentGetTimeOfProjectionNullRequest();
+        } catch (Request.InvalidRequestException e) {
+            buyTicketPresenter.presentInvalidGetTimeOfProjection(request);
+        }
+
+    }
+
+    @Override
+    public void confirmSeatsRandom() {
+        buyTicketPresenter.confirmSeatsRandom();
+    }
+
+    @Override
+    public void confirmSeatsSpecific() {
+        buyTicketPresenter.confirmSeatsSpecific();
+    }
+
+    @Override
+    public void setProjection(SetProjectionRequest request) {
```

Comment:
Sending a `set` message to this class should imply the intent of save the state of the projection somehow, you are just retrieving a projection here